Applications/demos/rocket_test/src/main.cpp

```diff
@@ -104,6 +104,20 @@ class dce : public Rocket::Core::ElementDocument {
   }
 };
 
+class evt_listener_instancer : public Rocket::Core::EventListenerInstancer {
+  virtual Rocket::Core::EventListener* InstanceEventListener(
+      const Rocket::Core::String& value, Rocket::Core::Element* element) {
+    (void)value;
+    (void)element;
+    return NULL;
+  }
+
+  /// Destroys the instancer.
+  virtual void Release() {
+    delete this;
```

Comment:
Is this not considered bad practice? Deleting an object from within itself

Reply:
Yes! But this is how libRocket expects to work :(. All of the objects that inherit from LibRocket have something similar. I think I Can fix it, but it will be a large change in the future.